File: java/Pascal's Triangle.java
public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        if(numRows < 1) return result;
        List<Integer> pre = new ArrayList<Integer>();
        pre.add(1);
        result.add(pre);
        for ( int i = 2; i <= numRows; i++){
            List<Integer> cur = new ArrayList<Integer>();
            cur.add(1);
            for(int j = 1; j<i-1;j++){
                cur.add(pre.get(j-1)+pre.get(j));
            }
            cur.add(1);
            result.add(cur);
            pre = cur;
        }
        return result;
        
        
    }

File: python/Pascal's Triangle.py
result=[]
        if numRows <1:
            return result
        pre = [1]
        result.append(pre)
        for i in range(2,numRows+1):
            cur=[1]
            for j in range(1, i-1):
                cur.append(pre[j]+pre[j-1])
            cur.append(1)
            result.append(cur)
            pre=cur
        return result
